File: backend/BusinessService/Common/Bases/Converter/CmConerterBase.cs
// *******************************************************************
// 業務名称　: 健康管理システム
// 機能概要　: ベースロジック
// 　　　　　　画面項目からDB項目に変換
// 作成日　　: 2022.12.12
// 作成者　　: 屠
// 変更履歴　:
// *******************************************************************
namespace BCC.Affect.Service
{
    public class CmConerterBase
    {
        /// <summary>
        /// 検索入力欄(通常)
        /// </summary>
        public static string? ToLikeStr(string? value)
        {
            if (string.IsNullOrEmpty(value))
            {
                return null;
            }
            return $"%{value}%";
        }
        /// <summary>
        /// 検索入力欄(カナ)
        /// </summary>
        public static string? ToKanaLikeStr(string? value)
        {
            if (string.IsNullOrEmpty(value))
            {
                return null;
            }

            return $"%{DaConvertUtil.ToKana(value)}%";
        }
        /// <summary>
        /// 検索入力欄(氏名カナ)
        /// </summary>
        public static string? ToKnameLikeStr(string? kname)
        {
            if (!string.IsNullOrEmpty(kname))
            {
                //カタカナへ変換
                kname = DaConvertUtil.ToKana(kname);
                //清音化
                kname = DaConvertUtil.ToSeion(kname);
                //%変換
                kname = kname.Replace("％", "%");
                return $"{kname}%";
            }
            else
            {
                return null;
            }
        }
    }
}

File: backend/BusinessService/Common/Bases/Converter/CmConerterBase.vb
' *******************************************************************
' 業務名称　: 互助防疫システム
' 機能概要　: ベースロジック
' 　　　　　　画面項目からDB項目に変換
' 作成日　　: 2024.07.12
' 作成者　　: 
' 変更履歴　:
' *******************************************************************
Namespace JBD.GJS.Service
    Public Class CmConerterBase
        ''' <summary>
        ''' 検索入力欄(通常)
        ''' </summary>
        Public Shared Function ToLikeStr(value As String) As String
            If String.IsNullOrEmpty(value) Then
                Return Nothing
            End If
            Return $"%{value}%"
        End Function
        ''' <summary>
        ''' 検索入力欄(カナ)
        ''' </summary>
        Public Shared Function ToKanaLikeStr(value As String) As String
            If String.IsNullOrEmpty(value) Then
                Return Nothing
            End If

            Return $"%{Db.DaConvertUtil.ToKana(value)}%"
        End Function
        ''' <summary>
        ''' 検索入力欄(氏名カナ)
        ''' </summary>
        Public Shared Function ToKnameLikeStr(kname As String) As String
            If Not String.IsNullOrEmpty(kname) Then
                'カタカナへ変換
                kname = Db.DaConvertUtil.ToKana(kname)
                '清音化
                kname = Db.DaConvertUtil.ToSeion(kname)
                '%変換
                kname = kname.Replace("％", "%")
                Return $"{kname}%"
            Else
                Return Nothing
            End If
        End Function
    End Class
End Namespace
